### DISCUSSION TITLE
Prevent invalidate while using enhance action

### DISCUSSION THREAD

**Discussion by PhanDungTri** [sveltejs/kit#10702] [Q&A] [answered]
How can I instruct Svelte **NOT** to invalidate the page data after the form submission? So that the `load` function of that page will not rerun.

**Comment by teemingc** [comment, marked as answer]
https://kit.svelte.dev/docs/form-actions#progressive-enhancement-use-enhance

> To customise the behaviour, you can provide a 
SubmitFunction
 that runs immediately before the form is submitted, and (optionally) returns a callback that runs with the 
ActionResult
. Note that if you return a callback, the default behavior mentioned above is not triggered.

```svelte

<form
    method="POST"
    use:enhance={({ formElement, formData, action, cancel, submitter }) => {
        // `formElement` is this `<form>` element
        // `formData` is its `FormData` object that's about to be submitted
        // `action` is the URL to which the form is posted
        // calling `cancel()` will prevent the submission
        // `submitter` is the `HTMLElement` that caused the form to be submitted

        return async ({ result, update }) => {
            // `result` is an `ActionResult` object
            // `update` is a function which triggers the default logic that would be triggered if this callback wasn't set
        };
    }}
>
```
